indico/modules/events/layout/util.py

```diff
@@ -308,15 +308,44 @@ def is_menu_entry_enabled(entry_name, event):
     return get_menu_entry_by_name(entry_name, event).is_enabled
 
 
+@dataclass(frozen=True)
+class ConferenceTheme:
+    """
+    Holds the values of a given conference theme to be used by Indico.
+
+    Required:
+    - ``name``     -- string indicating the internal name used for the stylesheet which will be
+                      stored when the theme is selected in an event.
+    - ``css_path`` -- string indicating the relative location of the CSS file
+    - ``title``    -- string indicating the title displayed to the user when selecting the theme.
+
+    Optional:
+    - ``js_path``  -- string indicating the relative location for a simple, static javascript file
+    """
+
+    name: str
+    css_path: str
+    title: str
+    js_path: str = None
+
+
 def get_plugin_conference_themes():
     data = values_from_signal(signals.plugin.get_conference_themes.send(), return_plugins=True)
-    return {':'.join((plugin.name, name)): (path, title) for plugin, (name, path, title) in data}
+    return {
+        ':'.join((plugin.name, (theme_info.name
+                                if isinstance(theme_info, ConferenceTheme)  # backwards compatibility check
+                                else ConferenceTheme(*theme_info).name))
```

Comment:
I would probably move the legacy handling out of  the return to avoid the double code (and double class instantiation in the legacy case)

Reply:
How about this?


```py
def get_plugin_conference_themes():
    data = values_from_signal(signals.plugin.get_conference_themes.send(), return_plugins=True)
    data = [plugin, (theme_info if isinstance(theme_info, ConferenceTheme) else ConferenceTheme(*theme_info))
            for plugin, theme_info in data]
    return {f'{plugin.name}:{theme.name}': theme for plugin, theme in data}
```

Reply:
@ThiefMaster unfortunately that syntax, as written is throwing errors.

Reply:
Just missing some parentheses, I didn't test it.

```python
def get_plugin_conference_themes():
    data = values_from_signal(signals.plugin.get_conference_themes.send(), return_plugins=True)
    data = [(plugin, (theme_info if isinstance(theme_info, ConferenceTheme) else ConferenceTheme(*theme_info)))
            for plugin, theme_info in data]
    return {f'{plugin.name}:{theme.name}': theme for plugin, theme in data}
```

Reply:
`   data = values_from_signal(signals.plugin.get_conference_themes.send(), return_plugins=True)
    for plugin, theme_info in data:
        if not isinstance(theme_info, ConferenceTheme):
            theme_info = ConferenceTheme(*theme_info)
        return {f'{plugin.name}:{theme_info.name}': theme_info}`

This is how my brain logically tells me to do it, but like I stated before, it only shows 1 theme if multiple are pressed for some reason

Reply:
Yes, the `return` aborts the loop early... try the snippet from the last comment, I think it does exactly what's needed.

Reply:
Sorry, you snuck in some replies while I was coding this lol. Yeah I figured out the parenthesis